x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_overview_logic.test.ts

```diff
@@ -0,0 +1,123 @@
+};
+
+const DEFAULT_CRAWL_RULE: CrawlRule = {
+  id: '-',
+  policy: CrawlerPolicies.Allow,
+  rule: CrawlerRules.regex,
+  pattern: '.*',
+};
+
+describe('CrawlerOverviewLogic', () => {
+  const { mount } = new LogicMounter(CrawlerOverviewLogic);
+  const { http } = mockHttpValues;
+  const { flashAPIErrors } = mockFlashMessageHelpers;
+
+  beforeEach(() => {
+    mount();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('has expected default values', () => {
+    expect(CrawlerOverviewLogic.values).toEqual(DEFAULT_VALUES);
+  });
+
+  describe('actions', () => {
+    describe('setCrawlerData', () => {
+      it('should set all received data as top-level values', () => {
+        const crawlerData = {
+          domains: [
+            {
+              id: '507f1f77bcf86cd799439011',
+              createdOn: 'Mon, 31 Aug 2020 17:00:00 +0000',
+              url: 'moviedatabase.com',
+              documentCount: 13,
+              sitemaps: [],
+              entryPoints: [],
+              crawlRules: [],
+              defaultCrawlRule: DEFAULT_CRAWL_RULE,
+            },
+          ],
+        };
+
+        CrawlerOverviewLogic.actions.setCrawlerData(crawlerData);
+
+        expect(CrawlerOverviewLogic.values).toEqual({
+          ...DEFAULT_VALUES,
+          ...crawlerData,
+        });
+      });
+    });
+  });
+
+  describe('listeners', () => {
+    describe('fetchCrawlerData', () => {
+      it('calls setCrawlerData with retrieved data that has been converted from server to client', async () => {
+        jest.spyOn(CrawlerOverviewLogic.actions, 'setCrawlerData');
+
+        http.get.mockReturnValue(
+          Promise.resolve({
+            domains: [
+              {
+                id: '507f1f77bcf86cd799439011',
+                name: 'moviedatabase.com',
+                created_on: 'Mon, 31 Aug 2020 17:00:00 +0000',
+                document_count: 13,
+                sitemaps: [],
+                entry_points: [],
+                crawl_rules: [],
+              },
+            ],
+          })
+        );
+        CrawlerOverviewLogic.actions.fetchCrawlerData();
+        await nextTick();
+
+        expect(http.get).toHaveBeenCalledWith('/api/app_search/engines/some-engine/crawler');
+        expect(CrawlerOverviewLogic.actions.setCrawlerData).toHaveBeenCalledWith({
+          domains: [
+            {
+              id: '507f1f77bcf86cd799439011',
+              createdOn: 'Mon, 31 Aug 2020 17:00:00 +0000',
+              url: 'moviedatabase.com',
+              documentCount: 13,
+              sitemaps: [],
+              entryPoints: [],
+              crawlRules: [],
+            },
+          ],
+        });
+      });
+
+      it('calls flashApiErrors when there is an error', async () => {
+        http.get.mockReturnValue(Promise.reject());
+        CrawlerOverviewLogic.actions.fetchCrawlerData();
+        await nextTick();
+
+        expect(flashAPIErrors).toHaveBeenCalled();
+      });
```

Comment:
For consistency, or maybe just a more thorough test, I'd assert that the error is also passed to `flashApiErrors`

```suggestion
      it('calls flashApiErrors when there is an error', async () => {
        http.get.mockReturnValue(Promise.reject('error'));
        CrawlerOverviewLogic.actions.fetchCrawlerData();
        await nextTick();

        expect(flashAPIErrors).toHaveBeenCalledWith('error');
      });
```

Reply:
I've addressed this in 18b0a91